models/world_oecd_piaac/code/metadata.py

```
@@ -0,0 +1,722 @@
+
+    # Key on url: get_raw_data_sources returns `name`, which is the Portuguese
+    # name, so matching on the English one never hits and every re-run creates a
+    # fresh set of sources. The url is stable and unique per source.
+    existing_sources: dict[str, str] = {}
+    try:
+        for source in (
+            tool(BD.get_raw_data_sources)(dataset_slug=DATASET_SLUG, env=env)
+            or []
+        ):
+            existing_sources.setdefault(source.get("url"), source.get("id"))
+    except Exception as exc:
```

Ruff (0.16.2)

[warning] Do not catch blind exception: `Exception`

(BLE001)